integrationtest/backend/elasticsearch/src/test/java/org/hibernate/search/integrationtest/backend/elasticsearch/testsupport/util/ElasticsearchTckBackendFeatures.java

```diff
@@ -6,31 +6,46 @@
  */
 package org.hibernate.search.integrationtest.backend.elasticsearch.testsupport.util;
 
+import static org.hibernate.search.util.impl.integrationtest.backend.elasticsearch.dialect.ElasticsearchVersionUtils.isActualVersionBetween;
+import static org.hibernate.search.util.impl.integrationtest.backend.elasticsearch.dialect.ElasticsearchVersionUtils.isActualVersionBetweenIncluding;
+import static org.hibernate.search.util.impl.integrationtest.backend.elasticsearch.dialect.ElasticsearchVersionUtils.isActualVersionEquals;
+import static org.hibernate.search.util.impl.integrationtest.backend.elasticsearch.dialect.ElasticsearchVersionUtils.isActualVersionLess;
+import static org.hibernate.search.util.impl.integrationtest.backend.elasticsearch.dialect.ElasticsearchVersionUtils.isActualVersionLessOrEquals;
+import static org.hibernate.search.util.impl.integrationtest.backend.elasticsearch.dialect.ElasticsearchVersionUtils.isOpensearchDistribution;
+
 import java.math.BigDecimal;
 import java.math.BigInteger;
 
+import org.hibernate.search.backend.elasticsearch.ElasticsearchVersion;
 import org.hibernate.search.engine.search.common.SortMode;
 import org.hibernate.search.engine.spatial.GeoPoint;
 import org.hibernate.search.integrationtest.backend.tck.testsupport.util.TestedFieldStructure;
+import org.hibernate.search.util.impl.integrationtest.backend.elasticsearch.ElasticsearchTestHostConnectionConfiguration;
 import org.hibernate.search.util.impl.integrationtest.backend.elasticsearch.dialect.ElasticsearchTestDialect;
 import org.hibernate.search.integrationtest.backend.tck.testsupport.util.TckBackendFeatures;
 
 class ElasticsearchTckBackendFeatures extends TckBackendFeatures {
 
-	private final ElasticsearchTestDialect dialect;
+	private final ElasticsearchVersion actualVersion;
 
-	ElasticsearchTckBackendFeatures(ElasticsearchTestDialect dialect) {
-		this.dialect = dialect;
+	ElasticsearchTckBackendFeatures() {
+		this.actualVersion = ElasticsearchTestDialect.getActualVersion();
 	}
 
 	@Override
 	public boolean geoPointIndexNullAs() {
-		return dialect.supportsGeoPointIndexNullAs();
+		if ( isActualVersionLess( ElasticsearchTestDialect.getActualVersion(), "elastic:6.3.0" ) ) {
+			return false;
+		}
+		return true;
```

Comment:
Yeah... I'd usually go with simply returning the expression instead of this if/return/else/return thing. If that's a code style - let's make it consistent then. I've made it like it here as we wanted to be more verbose and clear on what's happening for particular versions, and this looked more human-friendly 😄.